tokio/src/runtime/builder.rs

```diff
@@ -1522,11 +1522,16 @@ impl Builder {
     }
 
     #[cfg(tokio_unstable)]
-    fn build_current_thread_local_runtime(&mut self) -> io::Result<LocalRuntime> {
+    fn build_current_thread_local_runtime(
+        &mut self,
+        opts: LocalOptions,
+    ) -> io::Result<LocalRuntime> {
         use crate::runtime::local_runtime::LocalRuntimeScheduler;
 
         let tid = std::thread::current().id();
 
+        self.before_park = opts.before_park;
```

Comment:
IMO this should be done only if `opts.on_xyz_abc` is `Some`.

Use case: 
```rust
let opts = LocalOptions::default();
// no on_before_park/on_after_unpark for opts !

Builder::new_current_thread()
    .enable_time()
    .on_before_park(...)  // 1
    .build_local(opts)    // 2
    .unwrap();
```

Currently the user application sets `on_before_park` callback at 1) and then 2) will silently wipe them